src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -54,13 +54,20 @@ public void Main(string[] args)
                 shutdownHandle.Set();
             });
 
-            var ignored = Task.Run(() =>
+            var appLifeline = host.ApplicationServices.GetRequiredService<IApplicationLifetime>();
+            if (appLifeline.ApplicationStarted == null)
```

Comment:
Yes, by design because that's how you overwrite the default behavior. You should also be able to overwrite the message.